qualifier_non_outermost_array diagnostic is too strict

After upgrading zig to version 0.16, when using the Accelerate library, I encountered the following error:
```c
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vImage.framework/Headers/Transform.h:1255:137: error: type qualifier used in non-outermost array type
VIMAGE_PF vImage_Error vImageLookupTable_Planar8toPlanar96( const vImage_Buffer *src, const vImage_Buffer *dest,  const Pixel_FFFF table[256], vImage_Flags flags ) VIMAGE_NON_NULL(1,2,3) API_AVAILABLE(macos(10.9), ios(7.0), watchos(1.0), tvos(7.0));
                                                                                                                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vImage.framework/Headers/Transform.h:1299:138: error: type qualifier used in non-outermost array type
VIMAGE_PF vImage_Error vImageLookupTable_Planar8toPlanar128( const vImage_Buffer *src, const vImage_Buffer *dest,  const Pixel_FFFF table[256], vImage_Flags flags ) VIMAGE_NON_NULL(1,2,3) API_AVAILABLE(macos(10.9), ios(7.0), watchos(1.0), tvos(7.0));
                                                                                                                                         ^
error: process exited with error code 1
```
 refer to https://ziggit.dev/t/zig-0-16-macos-translate-c-issue/14945 for the relevant discussion

This is because the qualifier_non_outermost_array check is too strict. Apple's vImage uses const Pixel_FFFF table[256], which should be accepted. Currently, this syntax causes arocc to report an error.

--- a/src/aro/Parser.zig
+++ b/src/aro/Parser.zig
@@ -3577,7 +3577,7 @@ const Declarator = struct {
                     if (elem_array_ty.len == .static) {
                         try p.err(source_tok, .static_non_outermost_array, .{});
                     }
-                    if (elem_qt.isQualified()) {
+                    if (elem_qt.isQualified() and elem_qt.type(p.comp) != .typedef) {
                         try p.err(source_tok, .qualifier_non_outermost_array, .{});
                     }
                 }

--- a/test/cases/typedef array qualifiers.c
+++ b/test/cases/typedef array qualifiers.c
@@ -1,0 +1,11 @@
+typedef float Pixel_FFFF[4];
+const Pixel_FFFF table[256];
+volatile Pixel_FFFF other_table[2];
+
+void invalid(int a[3][const 4]);
+
+/** manifest:
+syntax
+
+typedef array qualifiers.c:5:19: error: type qualifier used in non-outermost array type
+*/

